### DISCUSSION TITLE
Grouping blog posts in directories

### DISCUSSION THREAD

**Discussion by thosgood** [nanoc/nanoc#1617] [Get help] [answered]
I currently use nanoc for my website and blog, with all my blog posts stored in a single folder, like

```
content/
  blog/
    2020-01-01-first-post.md
    2020-02-01-second-post.md
    ...
```

and I keep all my images in a separate directory: `content/assets/post-images`. Having recently used blogdown for a different project, I really like the approach there where a post consists of a *directory* containing an `index.Rmd` file as well as any images etc. that also make up the content.

To get this to work in nanoc, I know I need to change my `Rules` file, which currently has

```
compile '/blog/*.md' do
  filter :pandoc, args: [
    :katex
  ]
  layout '/Post.*'
  filter :add_toc
  y,m,d,slug = /([0-9]+)\-([0-9]+)\-([0-9]+)\-([^\/]+).md/.match(item.identifier).captures
  write "/blog/#{y}/#{m}/#{d}/#{slug}/index.html"
end
```

but I haven't been able to figure out how exactly to modify this to get what I want.

Has anybody already implemented such an approach using nanoc? I'd love to hear more about it, or to get some advice!

**Comment by 0x1eef** [comment, marked as answer]
With an approach like this, it should be straight forward:

"First post"
* `./content/blog/2022-01-01/first-post/index.md`
* `./content/blog/2022-01-01/first-post/images/foo.png`

"Second post"
* `./content/blog/2022-02-01/second-post/index.md`
* `./content/blog/2022-02-01/second-post/images/bar.png`

"Rules"
```ruby
compile "/blog/*/*/index.md" do
  filter :foo
  filter :bar 
  write(item.identifier.to_s)
end

compile "/blog/*/*/images/*.png" do 
  write(item.identifier.to_s)
end
```

Then you can use relative paths from `index.md` (`<img src="images/foo.png">`, etc). Another advantage is that the source layout is closer to the build layout, and I think that makes it easier to reason about. I didn't test this though, it might need some tweaks.

**Reply by thosgood** [reply]
amazing, thank you! I changed this slightly (since I use `/blog/YYYY-MM-DD-slug` instead of `/blog/YYYY-MM-DD/slug`) and it works perfectly (for anybody else reading this, you can see my current version at https://github.com/thosgood/thosgood.github.io/blob/main/Rules)

**Reply by denisdefreyne** [reply]
If you want, you can shorten this a bit further with `passthrough`:

```ruby
passthrough "/blog/**/images/*.png"
```
